### PR TITLE
MINOR: add docs for KIP-680

### PR DESCRIPTION
**Pull request by mjsax**
Call for review @bbejeck.

Follow up PR to #9660 for 2.8.0 release.

### CODE REVIEW

**Review by bbejeck — APPROVED** [review]
LGTM

File: docs/streams/upgrade-guide.html [hunk 1]
```diff
@@ -121,6 +121,12 @@ <h3><a id="streams_api_changes_280" href="#streams_api_changes_280">Streams API
         the constructor, such as when using the console consumer. <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-659%3A+Improve+TimeWindowedDeserializer+and+TimeWindowedSerde+to+handle+window+size">KIP-659</a>
         has more details.
     </p>
+    <p>
+        To simplify testing, two new constructors are added to <code>TopologyTestDriver</code> class, that both don't
```
Comment (JimGalasyn) [inline docs/streams/upgrade-guide.html]:
```suggestion
        To simplify testing, two new constructors that don't require a <code>Properties</code> parameter have been added to the <code>TopologyTestDriver</code> class.
```

File: docs/streams/upgrade-guide.html [hunk 2]
```diff
@@ -121,6 +121,12 @@ <h3><a id="streams_api_changes_280" href="#streams_api_changes_280">Streams API
...
+        required to pass in a <code>Properties</code> parameter. Furthermore, even if <code>Properties</code> are passed
```
Comment (JimGalasyn) [inline docs/streams/upgrade-guide.html]:
```suggestion
        required to pass in a <code>Properties</code> parameter. If <code>Properties</code> are passed
```

**Review by JimGalasyn — APPROVED** [review]
LGTM, with a couple of suggestions.

File: docs/streams/upgrade-guide.html [hunk 3]
```diff
@@ -121,6 +121,12 @@ <h3><a id="streams_api_changes_280" href="#streams_api_changes_280">Streams API
         the constructor, such as when using the console consumer. <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-659%3A+Improve+TimeWindowedDeserializer+and+TimeWindowedSerde+to+handle+window+size">KIP-659</a>
         has more details.
     </p>
+    <p>
+        To simplify testing, two new constructors that don't require a <code>Properties</code> parameter have been added to the <code>TopologyTestDriver</code> class.
```
Comment (mjsax) [inline docs/streams/upgrade-guide.html]:
```suggestion
        To simplify testing, two new constructors that don't require a <code>Properties</code> parameter have been added
```

File: docs/streams/upgrade-guide.html [hunk 4]
```diff
@@ -121,6 +121,12 @@ <h3><a id="streams_api_changes_280" href="#streams_api_changes_280">Streams API
...
+        required to pass in a <code>Properties</code> parameter. If <code>Properties</code> are passed
```
Comment (mjsax) [inline docs/streams/upgrade-guide.html]:
```suggestion
        to the <code>TopologyTestDriver</code> class. If <code>Properties</code> are passed
```

**Pull request merged**